src/google/protobuf/compiler/csharp/csharp_primitive_field.cc

```diff
@@ -63,42 +63,60 @@ PrimitiveFieldGenerator::~PrimitiveFieldGenerator() {
 }
 
 void PrimitiveFieldGenerator::GenerateMembers(io::Printer* printer) {
-  // TODO(jonskeet): Work out whether we want to prevent the fields from ever being
-  // null, or whether we just handle it, in the cases of bytes and string.
-  // (Basically, should null-handling code be in the getter or the setter?)
+  
+  // Note: in multiple places, this code assumes that all fields
+  // that support presence are either nullable, or use a presence field bit.
+  // Fields which are oneof members are not generated here; they're generated in PrimitiveOneofFieldGenerator below.
+  // Extensions are not generated here either.
+
+
+  // Proto2 allows different default values to be specified. These are retained via static fields. They
+  // don't particularly need to be, but we don't need to change that. In Proto3 the default value we don't
```

Comment:
Just wanted to verify that fixing this wouldn't be an API-breaking change?

It looks like these static variables are private, so I suspect no. But if there are any API breaks we'd want to do, it seems like we should do them now, while the proto2 API is still considered experimental.

Perhaps we should review the C# API for proto2 again before we remove its experimental status and document it.

Reply:
I haven't thought it through in massive detail, but no, I don't *think* that it would be a breaking change at all. (It certainly wouldn't be a public API surface breaking change - it would be a matter of subtle semantic changes if anything.)

Definitely happy to review the C# API for proto2 again, yes.